pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -122,3 +127,36 @@ func retainJobSelectorFields(desired, observed *unstructured.Unstructured) (*uns
 	}
 	return desired, nil
 }
+
+const IndexKeyHPAScaleTargetRef = "spec.scaleTargetRef"
+
+func retainDeploymentFields(cli client.Client) retentionInterpreter {
```

Comment:
I have another idea. We should write a `controller` to watch the creation of HPA. Once there is a change in HPA, I will add a `label` to the target resource pointed by HPA. 


Then we only need to check if the resource has this specific label here to determine whether we need to perform the `retain` operation.

Reply:
yes, we have considered this method, it may not better than this method

Reply:
labels method may unable to resolve all scenes, follow my steps, e.g:

create `HPA` in karmada -> propagate `HPA` to `member1` -> [`hpa ctrl` find `deployement` failed] -> create `deployment` in karmada -> propagate `deployment` to member1 -> `HPA` in member1 updated (currentReplicas set to 1) -> `HPA` in karmada updated (currentReplicas set to 1) -> [`hpa ctrl` called, add labels to `deployment`] -> manually delete the labels in `deployment` -> add stress to `deployment` in member1 -> `HPA` in member1 add `deployment` replicas -> `deployment` in karmada control `deployment` in member1, replicas back to 1 -> the `currentReplicas` filed in `HPA` keep unchange -> [`hpa ctrl` not called, since `HPA` unchanged]

***

updated：it seems my e.g is wrong. Since we have collect `desiredReplicas` other than `currentReplicas`, when "add stress to `deployment` in member1", though `currentReplicas` may unchanged, but `desiredReplicas` would change.

Reply:
> though currentReplicas may unchanged, but desiredReplicas would change.


Yes.

Reply:
> manually delete the labels in deploymen

Even if you manually delete the labels in deployment, it can be added again by `hpa ctrl`.